### PR TITLE
Correct SOU_TRN vram data transfer format

### PR DESCRIPTION
**Pull request by leinacc**
Clear up that there can be >1 data transfer packet

### CODE REVIEW

File: src/SGB_Command_Sound.md [hunk 1]
```diff
@@ -44,16 +44,21 @@ Used to transfer sound code or data to SNES Audio Processing Unit memory
  1-F   Not used (zero)
 ```
 
-The sound code/data is sent by VRAM-Transfer (4 KBytes).
+The sound code/data is sent by VRAM-Transfer (4 KBytes), by optionally specifying some `data transfer` packets from the SNES to its APU, and ending with a `jump` packet.
```
Comment (ISSOtm) [inline src/SGB_Command_Sound.md:47]:
```suggestion
The sound code/data is sent by [VRAM transfer](<#VRAM Transfers>) as a contiguous list of "packets".
```

File: src/SGB_Command_Sound.md [hunk 2]
```diff
@@ -44,16 +44,21 @@ Used to transfer sound code or data to SNES Audio Processing Unit memory
...
+
 All 16-bit values are little-endian.
 
+Data transfer packet format:
 ```
  000-001  Size of transfer data
  002-003  Destination address in S-APU RAM (typically $2B00, see below)
  004-XXX  Data to be transferred
```
Comment (ISSOtm) [inline src/SGB_Command_Sound.md:55]:
```suggestion
 00-01  Size of data below (XX); if zero, this is instead a jump packet
 02-03  Destination address in S-APU RAM (typically $2B00, see below)
 04-XX  Data to be transferred
```

File: src/SGB_Command_Sound.md [hunk 3]
```diff
@@ -44,16 +44,21 @@ Used to transfer sound code or data to SNES Audio Processing Unit memory
...
- X+1-X+2  "End marker" (???), should be $0000
- X+3-X+4  S-APU jump address, should be $0400
- X+5-FFF  Remaining bytes ignored
+```
+
+Jump packet format:
+```
+ 000-001  Should be $0000
+ 002-003  S-APU jump address, should be $0400 to safely restart the built-in SGB BIOS' N-SPC sound engine
```
Comment (ISSOtm) [inline src/SGB_Command_Sound.md:61]:
```suggestion
 00-01  Must be $0000
 02-03  S-APU jump address, use $0400 to safely restart the built-in SGB BIOS' N-SPC sound engine
```

**Review by ISSOtm — CHANGES_REQUESTED** [review]
Right, that makes a lot more sense. Thanks!

**Comment by leinacc** [comment]
That is a lot clearer, thanks!

File: src/SGB_Command_Sound.md [hunk 4]
```diff
@@ -44,16 +44,21 @@ Used to transfer sound code or data to SNES Audio Processing Unit memory
...
-The sound code/data is sent by VRAM-Transfer (4 KBytes).
+The sound code/data is sent by [VRAM transfer](<#VRAM Transfers>) as a contiguous list of "packets".
+
 All 16-bit values are little-endian.
 
+Data transfer packet format:
+```
+ 00-01  Size of data below (XX); if zero, this is instead a jump packet
+ 02-03  Destination address in S-APU RAM (typically $2B00, see below)
+ 04-XX  Data to be transferred
```
Comment (ISSOtm) [inline src/SGB_Command_Sound.md:55]: Wait, I'm not so sure anymore. Does `XX` include the 4-byte header? Probably not; then `04-XX` should become `004-N+4`, or something like that.

**Review by ISSOtm — CHANGES_REQUESTED** [review]

File: src/SGB_Command_Sound.md [hunk 5]
```diff
@@ -44,16 +44,21 @@ Used to transfer sound code or data to SNES Audio Processing Unit memory
...
+Jump packet format:
 ```
- 000-001  Size of transfer data
- 002-003  Destination address in S-APU RAM (typically $2B00, see below)
- 004-XXX  Data to be transferred
- X+1-X+2  "End marker" (???), should be $0000
- X+3-X+4  S-APU jump address, should be $0400
- X+5-FFF  Remaining bytes ignored
+ 00-01  Must be $0000
+ 02-03  S-APU jump address, use $0400 to safely restart the built-in SGB BIOS' N-SPC sound engine
```
Comment (ISSOtm) [inline src/SGB_Command_Sound.md:61]:
```suggestion
 0-1  Must be $0000
 2-3  S-APU jump address, use $0400 to safely restart the built-in SGB BIOS' N-SPC sound engine
```

File: src/SGB_Command_Sound.md [hunk 6]
```diff
@@ -44,16 +44,21 @@ Used to transfer sound code or data to SNES Audio Processing Unit memory
...
-The sound code/data is sent by VRAM-Transfer (4 KBytes).
+The sound code/data is sent by [VRAM transfer](<#VRAM Transfers>) as a contiguous list of "packets".
+
 All 16-bit values are little-endian.
 
+Data transfer packet format:
+```
+ 00-01  Size of data below (XX); if zero, this is instead a jump packet
+ 02-03  Destination address in S-APU RAM (typically $2B00, see below)
+ 04-N+4  Data to be transferred
```
Comment (ISSOtm) [inline src/SGB_Command_Sound.md:55]:
```suggestion
 0-1    Size of data below (N); if zero, this is instead a jump packet
 2-3    Destination address in S-APU RAM (typically $2B00, see below)
 4-N+4  Data to be transferred
```

**Review by ISSOtm — CHANGES_REQUESTED** [review]
Let's keep the alignment, then.

**Review by ISSOtm — APPROVED** [review]

File: src/SGB_Command_Sound.md [hunk 7]
```diff
@@ -44,16 +44,21 @@ Used to transfer sound code or data to SNES Audio Processing Unit memory
  1-F   Not used (zero)
 ```
 
-The sound code/data is sent by VRAM-Transfer (4 KBytes).
+The sound code/data is sent by [VRAM transfer](<#VRAM Transfers>) as a contiguous list of "packets".
```
Comment (avivace) [inline src/SGB_Command_Sound.md:47]: Should "data" be enough here instead of the potentially confusing "code and/or data"?

**Review by avivace — CHANGES_REQUESTED** [review]

File: src/SGB_Command_Sound.md (hunk 7 above)
Comment (ISSOtm) [reply src/SGB_Command_Sound.md:47]: Well, most applications of `SOU_TRN` will only upload custom data for Kankichi-kun, but it's possible to upload a custom sound driver as well. Hence "code and/or data".

Comment (avivace) [reply src/SGB_Command_Sound.md:47]: @ISSOtm okay, that makes sense. Thanks for the clarification.

**Pull request merged**